app/deploy/gcp/variables.tf

```diff
@@ -0,0 +1,34 @@
+variable "queue_id" {
+  type        = string
+  description = "The queue id for the worker"
+}
+
+variable "region" {
+  type        = string
+  description = "The region in which we're rolling out IaC"
+}
+
+variable "opentelemetry_collector_image" {
+  type        = string
+  description = "Address of the container image to use for the OpenTelemetry Collector"
+}
+
+variable "worker_image" {
+  type        = string
+  description = "Address of the container image to use for the worker"
+}
+
+variable "worker_groups" {
```

Comment:
You might have to explain this bit to me when we meet, like I think I get it, but it's still a bit fuzzy.

Reply:
The variable `worker_groups` is a map (so named key/value pairs) where the key is the name of the group -- whatever DNS-friendly name you want it to be -- and value is a custom object.

```
worker_groups = {
  my_group_a = { [object contents] }
  my_group_b = { [object contents] }
}
```

The object defined here has a few fields:
- `queue_id` Required.
- `instance_type` Required. The name of the underlying GCP instance type you want to launch.
- `cpus` Required, number of CPUs the instance type above will provide (this isn't automatically derived, but is passed to the worker config at runtime)
- `gpus` Optional. How many and what type of guest GPUs you want attached to the instance. This option is only valid if the instance is an N1-series instance, like `n1-standard-1`.
- `min_workers`/`max_workers` Optional. Just what they sound like :)

So an array of these looks like what is currently shown in the `variables.auto.tfvars` example. If there's anything that should still be clarified, I'm very happy to do so prior to our next meet!